File: Two Sum II - Leetcode 167/Two Sum II - Leetcode 167.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;

        while (l < r) {
            int sum = numbers[l] + numbers[r];
            if (sum == target) {
                return new int[] {l + 1, r + 1};
            } else if (sum < target) {
                l++;
            } else {
                r--;
            }
        }

        throw new IllegalArgumentException("No two sum solution");
    }
}

File: Two Sum II - Leetcode 167/Two Sum II - Leetcode 167.js
var twoSum = function(numbers, target) {
    let l = 0;
    let r = numbers.length - 1;

    while (l < r) {
        let sum = numbers[l] + numbers[r];
        if (sum === target) {
            return [l + 1, r + 1];
        } else if (sum < target) {
            l++;
        } else {
            r--;
        }
    }

    throw new Error("No two sum solution");
};
